Analysis workflow fails terminally when a vulnerability has an empty description ("Missing content")

### Current Behavior

reconcile-vuln-analysis-results fails terminally (6/6 attempts) whenever a newly identified vulnerability has an empty description, aborting the whole vuln-analysis workflow so findings are never persisted.

JdbcNotificationEmitter.validateRequiredFields rejects the generated GROUP_NEW_VULNERABILITY notification because its content field is blank. content is derived from the vulnerability description, and NVD publishes recently-assigned CVEs with an empty description (severity UNASSIGNED, sometimes an empty title as well). A missing description on an upstream record should not be able to fail an analysis workflow.

```
2026-07-27 14:51:58,565 WARN [ActivityTaskWorker] Activity failed terminally after 6 attempt(s) [activityName=reconcile-vuln-analysis-results, workflowRunId=019fa3a0-2efe-7908-aa5d-69b38845ad09, activityTaskAttempt=6]
java.lang.IllegalArgumentException: Invalid notification: level: LEVEL_INFORMATIONAL scope: SCOPE_PORTFOLIO group: GROUP_NEW_VULNERABILITY title: "New Vulnerability Identified on Project: [redacted : redacted]" ...
      at org.dependencytrack.notification.JdbcNotificationEmitter.emitAll(JdbcNotificationEmitter.java:111)
      at org.dependencytrack.notification.JdbcNotificationEmitter.emitAll(JdbcNotificationEmitter.java:83)
      at org.dependencytrack.notification.JdbiNotificationEmitter.emitAll(JdbiNotificationEmitter.java:33)
      at org.dependencytrack.vulnanalysis.ReconcileVulnAnalysisResultsActivity.lambda$reconcileFindings$5(ReconcileVulnAnalysisResultsActivity.java:567)
      at org.dependencytrack.vulnanalysis.ReconcileVulnAnalysisResultsActivity.reconcileFindings(ReconcileVulnAnalysisResultsActivity.java:513)
      at org.dependencytrack.vulnanalysis.ReconcileVulnAnalysisResultsActivity.execute(ReconcileVulnAnalysisResultsActivity.java:193)
      at org.dependencytrack.vulnanalysis.ReconcileVulnAnalysisResultsActivity.execute(ReconcileVulnAnalysisResultsActivity.java:91)
      at org.dependencytrack.dex.engine.ActivityTaskWorker.lambda$process$0(ActivityTaskWorker.java:117)
Caused by: java.lang.IllegalArgumentException: Missing content
      at org.dependencytrack.notification.JdbcNotificationEmitter.validateRequiredFields(JdbcNotificationEmitter.java:209)
      at org.dependencytrack.notification.JdbcNotificationEmitter.emitAll(JdbcNotificationEmitter.java:108)
      ... 23 common frames omitted
```

Decoding the NewVulnerabilitySubject protobuf from the failing payload shows the subject is otherwise well-formed; the vulnerability message carries an explicitly-set zero-length description:

field 3 (message, 190B):            # vulnerability
  field 1  = '187d20be-43dd-455b-b64c-bd3318dde4c2'
  field 2  = 'CVE-2026-53614'
  field 3  = 'NVD'
  field 5  = '[Local Privilege Escalation via LIBMOUNT_FORCE_MOUNT2 Environment Variable - nosuid/noexec Bypass in SUID mount(8)]'
  field 7  = <EMPTY, length 0>      # description
  field 14 = 'UNASSIGNED'

Confirmed against the database:

```
SELECT "VULNID", "SOURCE", "SEVERITY",
       COALESCE(LENGTH("TITLE"), -1) AS title_len,
       COALESCE(LENGTH("DESCRIPTION"), -1) AS desc_len
FROM "VULNERABILITY" WHERE "VULNID" IN ('CVE-2026-53613','CVE-2026-53614');
```

```text
     VULNID     | SOURCE |  SEVERITY  | title_len | desc_len
----------------+--------+------------+-----------+----------
 CVE-2026-53613 | NVD    | UNASSIGNED |        77 |        0
 CVE-2026-53614 | NVD    | UNASSIGNED |       115 |        0
(2 rows)
```

This is not limited to a single record. On our instance 32 vulnerabilities have an empty-string description, and a further 33,294 have NULL:

```text
 SOURCE  | null_desc | empty_desc
---------+-----------+------------
 OSV     |     33294 |          0
 UNKNOWN |         4 |         22
 NVD     |         0 |         10
(3 rows)
```

The UNKNOWN-source rows are Debian security-tracker entries (DSA-*, DLA-*, TEMP-*) that match base packages on any Debian-based image, so the reachable surface is broad. We have not established whether NULL descriptions trigger the same path as empty strings.

Notably, this reproduces with every notification rule disabled — which is the default state after a v4→v5 migration, since the migrator sets all migrated rules to ENABLED=FALSE. The notification is constructed and validated before rule routing, so a payload with no subscribers at all still aborts the analysis.

Impact: the BOM upload reports success to the client (BOM processed successfully, BOM import completed), then analysis dies at reconcile and findings are never written. A CI pipeline gating on Dependency-Track findings sees a green result with no vulnerabilities rather than an error, a silent false negative.

Workaround in use — backfilling the description from the title unblocks analysis, but the mirror may reset it on the next sync:

```
UPDATE "VULNERABILITY"
SET "DESCRIPTION" = COALESCE(NULLIF("TITLE", ''), 'No description available.')
WHERE "DESCRIPTION" = '';
```

### Steps to Reproduce

1. Run Dependency-Track v5 with the NVD vulnerability data source enabled and mirrored.
2. Ensure the vulnerability database contains a record with an empty description, e.g. a recently-assigned CVE such as CVE-2026-53613 / CVE-2026-53614, or force one:

`UPDATE "VULNERABILITY" SET "DESCRIPTION" = '' WHERE "VULNID" = '<some-cve>';`

3. Upload a CycloneDX SBOM containing a component that matches that vulnerability (e.g. pkg:deb/debian/bsdutils@2.41-5 or pkg:deb/debian/liblastlog2-2@2.41-5 on debian-13.6).
4. Watch the API server worker logs.

The BOM import completes, vuln-analysis starts, then reconcile-vuln-analysis-results fails 6 times with Missing content and the workflow never completes. No findings are persisted for the project.


### Expected Behavior

A vulnerability with an empty or missing description should not fail the analysis workflow. Either:

- fall back to a non-blank content when the description is empty (e.g. the vulnerability title, or a placeholder), or
- treat notification emission as best-effort for this activity — log and skip the invalid notification rather than aborting reconciliation, so findings are still persisted.

Failing the entire reconcile-vuln-analysis-results activity means the analysis result is lost, which is a much larger consequence than a dropped notification.

### Dependency-Track Version

5.x

### Browser

Google Chrome

### Checklist

- [x] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/main/CONTRIBUTING.md#filing-issues)
- [x] I have checked the [existing issues](https://github.com/DependencyTrack/dependency-track/issues) for whether this defect was already reported

--- a/notification/api/src/main/java/org/dependencytrack/notification/api/NotificationFactory.java
+++ b/notification/api/src/main/java/org/dependencytrack/notification/api/NotificationFactory.java
@@ -247,18 +247,9 @@ public final class NotificationFactory {
         }
         title += "]";
 
-        final String content;
-        if (vulnerability.hasDescription()) {
-            content = vulnerability.getDescription();
-        } else {
-            content = vulnerability.hasTitle()
-                    ? "%s: %s".formatted(vulnerability.getVulnId(), vulnerability.getTitle())
-                    : vulnerability.getVulnId();
-        }
-
         return newNotificationBuilder(SCOPE_PORTFOLIO, GROUP_VULNERABILITY_RETRACTED, LEVEL_INFORMATIONAL)
                 .setTitle(title)
-                .setContent(content)
+                .setContent(contentFor(vulnerability))
                 .setSubject(Any.pack(
                         VulnerabilityRetractedSubject.newBuilder()
                                 .setProject(project)
@@ -294,18 +285,9 @@ public final class NotificationFactory {
         }
         title += "]";
 
-        final String content;
-        if (vulnerability.hasDescription()) {
-            content = vulnerability.getDescription();
-        } else {
-            content = vulnerability.hasTitle()
-                    ? "%s: %s".formatted(vulnerability.getVulnId(), vulnerability.getTitle())
-                    : vulnerability.getVulnId();
-        }
-
         return newNotificationBuilder(SCOPE_PORTFOLIO, GROUP_NEW_VULNERABILITY, LEVEL_INFORMATIONAL)
                 .setTitle(title)
-                .setContent(content)
+                .setContent(contentFor(vulnerability))
                 .setSubject(Any.pack(
                         NewVulnerabilitySubject.newBuilder()
                                 .setProject(project)
@@ -603,6 +585,17 @@ public final class NotificationFactory {
                 .setScope(scope)
                 .setGroup(group)
                 .setLevel(level);
+    }
+
+    private static String contentFor(Vulnerability vulnerability) {
+        if (!vulnerability.getDescription().isBlank()) {
+            return vulnerability.getDescription();
+        }
+        if (!vulnerability.getTitle().isBlank()) {
+            return "%s: %s".formatted(vulnerability.getVulnId(), vulnerability.getTitle());
+        }
+
+        return vulnerability.getVulnId();
     }
 
 }

--- a/notification/api/src/test/java/org/dependencytrack/notification/api/NotificationFactoryTest.java
+++ b/notification/api/src/test/java/org/dependencytrack/notification/api/NotificationFactoryTest.java
@@ -18,11 +18,23 @@
  */
 package org.dependencytrack.notification.api;
 
+import org.dependencytrack.notification.proto.v1.AnalysisTrigger;
+import org.dependencytrack.notification.proto.v1.Component;
 import org.dependencytrack.notification.proto.v1.Notification;
+import org.dependencytrack.notification.proto.v1.Project;
+import org.dependencytrack.notification.proto.v1.Vulnerability;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.function.Supplier;
+import java.util.stream.Stream;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.dependencytrack.notification.api.NotificationFactory.createNewVulnerabilityNotification;
+import static org.dependencytrack.notification.api.NotificationFactory.createVulnerabilityRetractedNotification;
 import static org.dependencytrack.notification.api.NotificationFactory.newNotificationBuilder;
 import static org.dependencytrack.notification.proto.v1.Group.GROUP_BOM_CONSUMED;
 import static org.dependencytrack.notification.proto.v1.Group.GROUP_UNSPECIFIED;
@@ -30,6 +42,7 @@ import static org.dependencytrack.notification.proto.v1.Level.LEVEL_INFORMATIONA
 import static org.dependencytrack.notification.proto.v1.Level.LEVEL_UNSPECIFIED;
 import static org.dependencytrack.notification.proto.v1.Scope.SCOPE_PORTFOLIO;
 import static org.dependencytrack.notification.proto.v1.Scope.SCOPE_UNSPECIFIED;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
 
 class NotificationFactoryTest {
 
@@ -90,6 +103,58 @@ class NotificationFactoryTest {
                 .isThrownBy(() -> newNotificationBuilder(
                         SCOPE_PORTFOLIO, GROUP_BOM_CONSUMED, LEVEL_UNSPECIFIED))
                 .withMessage("Invalid level: LEVEL_UNSPECIFIED");
+    }
+
+    @ParameterizedTest
+    @MethodSource("vulnerabilityContentArguments")
+    void createNewVulnerabilityNotificationShouldPopulateContent(
+            Vulnerability vulnerability,
+            String expectedContent) {
+        final Notification notification =
+                createNewVulnerabilityNotification(
+                        Project.newBuilder().setName("acme-app").build(),
+                        Component.newBuilder().setName("acme-lib").build(),
+                        vulnerability,
+                        AnalysisTrigger.ANALYSIS_TRIGGER_BOM_UPLOAD);
+        assertThat(notification.getContent()).isEqualTo(expectedContent);
+    }
+
+    @ParameterizedTest
+    @MethodSource("vulnerabilityContentArguments")
+    void createVulnerabilityRetractedNotificationShouldPopulateContent(
+            Vulnerability vulnerability,
+            String expectedContent) {
+        final Notification notification =
+                createVulnerabilityRetractedNotification(
+                        Project.newBuilder().setName("acme-app").build(),
+                        Component.newBuilder().setName("acme-lib").build(),
+                        vulnerability);
+        assertThat(notification.getContent()).isEqualTo(expectedContent);
+    }
+
+    private static Stream<Arguments> vulnerabilityContentArguments() {
+        final Supplier<Vulnerability.Builder> vulnBuilder =
+                () -> Vulnerability.newBuilder().setVulnId("CVE-123").setSource("NVD");
+
+        return Stream.of(
+                arguments(
+                        vulnBuilder.get().setDescription("Some description").setTitle("Some title").build(),
+                        "Some description"),
+                arguments(
+                        vulnBuilder.get().setDescription("").setTitle("Some title").build(),
+                        "CVE-123: Some title"),
+                arguments(
+                        vulnBuilder.get().setTitle("Some title").build(),
+                        "CVE-123: Some title"),
+                arguments(
+                        vulnBuilder.get().setDescription("").setTitle("").build(),
+                        "CVE-123"),
+                arguments(
+                        vulnBuilder.get().setDescription("   ").build(),
+                        "CVE-123"),
+                arguments(
+                        vulnBuilder.get().build(),
+                        "CVE-123"));
     }
 
 }

